internal/certificates/certificate_subresource.go

```diff
@@ -0,0 +1,145 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package certificates
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+func CertificateSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:        schema.TypeString,
+				Description: "Domain of the resource certificate",
+				Computed:    true,
+			},
+			"expiration_status": {
+				Type:        schema.TypeString,
+				Description: "Certificate expiry status. One among: ACTIVE, ABOUT_TO_EXPIRE, EXPIRED",
+				Computed:    true,
+			},
+			"certificate_error": {
+				Type:        schema.TypeString,
+				Description: "Error if certificate cannot be fetched. Example: Status : NOT_TRUSTED, Message : Certificate Expired",
+				Computed:    true,
+			},
+			"is_installed": {
+				Type:        schema.TypeBool,
+				Description: "Whether the certificate is installed or not",
+				Computed:    true,
+			},
+			"issued_by": {
+				Type:        schema.TypeString,
+				Description: "The certificate authority that issued the certificate",
+				Computed:    true,
+			},
+			"issued_to": {
+				Type:        schema.TypeString,
+				Description: "To whom the certificate is issued",
+				Computed:    true,
+			},
+			"key_size": {
+				Type:        schema.TypeString,
+				Description: "The key size of the certificate",
+				Computed:    true,
+			},
```

Comment:
Are these computed or are the inputs to the cert resource?

Reply:
This whole subresource represents a certificate output as result of internal/external CA certificate replacement. All attributes are computed as a requirement of the Terraform SDK.